uxarray/grid/coordinates.py

```diff
@@ -244,6 +322,211 @@ def _construct_face_centroids(node_x, node_y, node_z, face_nodes, n_nodes_per_fa
     return _normalize_xyz(centroid_x, centroid_y, centroid_z)
 
 
+def haversine_distance(point1, point2):
+    """Calculate the great-circle distance between two points on a unit sphere
+    using the Haversine formula.
+
+    Parameters
+    ----------
+    point1 : tuple
+        A tuple containing the lon and lat of the first point.
+    point2 : tuple
+        A tuple containing the lon and lat of the second point.
+
+    Returns
+    -------
+    float
+        The distance between the two points on the unit sphere.
+    """
+    R = 1.0  # Radius of the Earth assumed to be 1 (unit sphere)
+    lon1, lat1 = np.radians(point1)
+    lon2, lat2 = np.radians(point2)
+
+    dlat = lat2 - lat1
+    dlon = lon2 - lon1
+
+    a = np.sin(dlat / 2) ** 2 + np.cos(lat1) * np.cos(lat2) * np.sin(dlon / 2) ** 2
+    c = 2 * np.arctan2(np.sqrt(a), np.sqrt(1 - a))
+
+    distance = R * c
+    return distance
+
+
+def circle_from_two_points(p1, p2):
```

Comment:
How about the wrap-around case where the great circle goes across the anti-meridian? Can you explain how you treat that case?

Also, this looks like a duplication of the haversine_distance, especially if we use the `_angle_of_two_vector ` function.